lib/multiplexer/proxyline.go

```diff
@@ -60,6 +60,38 @@ func (p *ProxyLine) String() string {
 	return fmt.Sprintf("PROXY %s %s %s %d %d\r\n", p.Protocol, p.Source.IP.String(), p.Destination.IP.String(), p.Source.Port, p.Destination.Port)
 }
 
+// Bytes returns on-the wire bytes representation of proxy line conforming to the proxy v2 protocol
+func (p *ProxyLine) Bytes() []byte {
+	var b []byte
+	b = append(b, proxyV2Prefix...)
+	b = append(b, (Version2<<4)|ProxyCommand)
+
+	switch p.Protocol {
+	case TCP4:
+		b = append(b, ProtocolTCP4)
+		b = append(b, 0, 12)
+		b = append(b, p.Source.IP.To4()...)
+		b = append(b, portToBytes(p.Source)...)
+		b = append(b, p.Destination.IP.To4()...)
+		b = append(b, portToBytes(p.Destination)...)
+	case TCP6:
+		b = append(b, ProtocolTCP6)
+		b = append(b, 36)
```

Comment:
Ditto, what does 36 mean?

Reply:
I've reworked this whole section to use structs instead, which avoids all these magic numbers